src/Dapr.Actors/Communication/ActorInvokeExceptionData.cs

```diff
@@ -0,0 +1,52 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+namespace Dapr.Actors
+{
+    using System;
+    using System.IO;
+    using System.Runtime.Serialization;
+    using System.Xml;
+    using Dapr.Actors;
+    using Microsoft.Extensions.Logging;
+
+    [DataContract(Name = "ActorInvokeExceptionData", Namespace = Constants.Namespace)]
+    internal class ActorInvokeExceptionData
+    {
+        private static readonly DataContractSerializer ActorInvokeExceptionDataSerializer = new DataContractSerializer(typeof(ActorInvokeExceptionData));
+
+        public ActorInvokeExceptionData(string type, string message)
+        {
+            this.Type = type;
+            this.Message = message;
+        }
+
+        [DataMember]
+        public string Type { get; private set; }
+
+        [DataMember]
+        public string Message { get; private set; }
+
+        internal static ActorInvokeExceptionData Deserialize(Stream buffer)
+        {
+            if ((buffer == null) || (buffer.Length == 0))
+            {
+                return null;
+            }
+
+            using var reader = XmlDictionaryReader.CreateBinaryReader(buffer, XmlDictionaryReaderQuotas.Max);
```

Comment:
You have to be careful here with usings on these readers. For instance I believe this class will dispose the underlying stream.

This isn't a problem if you're using memory stream, but with streams that actually do I/O you actually care about the lifetimes.

Reply:
This is one of the reasons why I want to limit the number and kind of serializers we're using. The S.T.Json API for instance is really simple, and doesn't bump into concerns like this.